### ISSUE TITLE
cache in VertexGrid does not work

### ISSUE THREAD

**Issue by rubencalje** [modflowpy/flopy#520]
When I run the following code for a model with a VertexGrid:

```
for cell2d in range(mf.modelgrid.ncpl):
    vert = mf.modelgrid.get_cell_vertices(cell2d)
```

This is much slower than when I run:

```
xv,yv,zv = mf.modelgrid.xyzvertices
for cell2d in range(mf.modelgrid.ncpl):
    vert = list(zip(xv[cell2d],yv[cell2d]))
```

I believe this is because the cache in the method of xyzvertices in VertexGrid is not working correctly, but I do not understand exactly how this cache works.

**Comment by spaulins-usgs** [comment]
You are correct that cache in the xyzvertices method is not working in the first loop. This is because "mf.modelgrid" creates a new VertexGrid every time you call it. The "mf" model object does not return a cached Grid, but constructs a new one every time you call it. This is by design since the spatial information in the model could potentially change in the dis/disv/disu packages between calls to "mf.modelgrid". If you rewrite the first loop like this:

mg = mf.modelgrid
for cell2d in range(mg.ncpl):
    vert = mg.get_cell_vertices(cell2d)

you will see some performance improvement. However, it still is not as fast as your second loop due to a second performance problem. The second problem has to do with flopy always returning a copy of the internal cache instead of the actual cache. Flopy does this because the cache it returns is not immutable. However, in the case of get_cell_vertices copying the cache is not necessary since the "list(zip(self.xvertices[cellid], self.yvertices[cellid]))" is already making a copy of the relevant data.

I have fixed the code not to make extra copies of the cache in this case. The fix makes the loop above as fast as your second loop. This fix is currently in a pull request and can be found here:

https://github.com/spaulins-usgs/flopy/tree/develop/flopy/discretization

The files that have been updated are grid.py, structuredgrid.py, and vertexgrid.py.
